controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/RoutingPolicies.java

```diff
@@ -153,10 +153,17 @@ private void removeObsoletePolicies(AllocatedLoadBalancers loadBalancers, Set<Ro
             LOGGER.log(LogLevel.WARNING, "Removing " + removalCandidates + ". Active load balancers " +
                                          activeLoadBalancers);
         }
+        removalCandidates.forEach(allPolicies::remove);
+
+        // Find CNAMEs in use
+        var cnamesUsed = allPolicies.stream()
+                .map(policy -> policy.endpointIn(controller.system()).dnsName())
+                .collect(Collectors.toSet());
+
         for (var policy : removalCandidates) {
             var dnsName = policy.endpointIn(controller.system()).dnsName();
-            controller.nameServiceForwarder().removeRecords(Record.Type.CNAME, RecordName.from(dnsName), Priority.normal);
-            allPolicies.remove(policy);
+            if ( ! cnamesUsed.contains(dnsName))
```

Comment:
When does this happen? `dnsName` should be unique per policy the policy itself is for a single cluster.

Reply:
Internal ticket updated with logs showing why this PR was made, it looked like removing an old policy for the same CNAME ended up removing the CNAME as well, even though a newer policy existed.